common/persistence/elasticsearch/es_visibility_store.go

```diff
@@ -604,6 +643,56 @@ func getSQLFromListRequest(request *p.ListWorkflowExecutionsByQueryRequest) stri
 	return sql
 }
 
+func getSQLFromListAllRequest(request *p.InternalListAllWorkflowExecutionsByTypeRequest) string {
+	sql := "select * from dummy "
+	var earliestTimeStr, latestTimeStr, timeRange, searchString, whereClause, orderByClause string
+
+	if !request.EarliestTime.IsZero() && !request.LatestTime.IsZero() {
+		earliestTimeStr = strconv.FormatInt(request.EarliestTime.UnixNano()-oneMicroSecondInNano, 10)
+		latestTimeStr = strconv.FormatInt(request.LatestTime.UnixNano()+oneMicroSecondInNano, 10)
+		timeRange = fmt.Sprintf("%s between %s and %s", es.StartTime, earliestTimeStr, latestTimeStr)
+		whereClause = addToWhereClause(whereClause, timeRange)
+	}
+
+	statusFilters := make([]string, 0)
```

Comment:
nit: preallocate the slice with the size to avoid extra allocation.
```suggestion
	statusFilters := make([]string, 0, len(request.StatusFilter))
```

Reply:
updated